world.go

```diff
@@ -79,9 +89,8 @@ func (w *World) RegisterComponent(c Component) ComponentID {
 		return id
 	}
 
-	nextID := ComponentID(len(w.factories))
-
-	factory := newComponentFactory(nextID)
+	nextId := atomic.AddUint64(w.nextEntityID, 1)
```

Comment:
is this a concurrency thing?

Reply:
Yeah, `atomic` is a thread-safe thing that allows you to add/subtract and read a uint/int from different threads without needing to explicitly guard it with a mutex